packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -913,15 +913,25 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                     return create404Response(request);
                 }
             } else if (requestParser.pathParts.length > 0) {
-                const wait =
-                    typeof request.headers?.wait === "boolean" ? request.headers.wait : undefined;
-
-                const dataStore = await this.getDataStore(id, wait);
-                const subRequest = requestParser.createSubRequest(1);
-                // We always expect createSubRequest to include a leading slash, but asserting here to protect against
-                // unintentionally modifying the url if that changes.
-                assert(subRequest.url.startsWith("/"), "Expected createSubRequest url to include a leading slash");
-                return dataStore.IFluidRouter.request(subRequest);
+                /**
+                 * If the request has "errorOnUnreferencedResources" header, we need to return an error if the data
```

Comment:
I'm not sure I like "errorOnUnreferencedResources" naming. I'd rather prefer to differentiate on internal vs. externa request. I.e. only one specific path would get "internal" behavior (resolving handles), but all external calls to ContainerRuntime.resolveHandle should get default "external" behavior. That would be ideal default, but Bohemia (given they do not use handles directly) can set same flag to differentiate.

Reply:
I don't like adding this by default to all external requests. I would rather have any callers that want this behavior to explicitly set this flag. IMO, we are adding this code to solve a very specific use case and I would not want to apply this to everyone. The app is in the best position to know if and in which scenario they want this behavior and they should set the flag in that path only.

Reply:
Sure, I think you are debating the meaning of having or not having a flag. I worry much more about the name - IMHO it should be about external vs. internal request, not about return unreferenced resource or not.

Reply:
Ah okay! I am open to naming suggestion :-).

Reply:
I'd prefer to see "internalRequest" or "externalRequest" to communicate intention, not behavior here.